### PR TITLE
[MkFit] round up size arguments to aligned_alloc upwards to align value.

### PR DESCRIPTION
**Pull request by osschar**
std::aligned_alloc calls were passing the desired allocated size, not rounded up to a multiple of align argument which seems to result in UB according to the standard.

All allocations in mkFit are aligned to cache-line (64) required for vector intrinsics.

This PR addresses #37096.

### CODE REVIEW

File: RecoTracker/MkFitCore/src/HitStructures.cc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #include "RecoTracker/MkFitCore/interface/HitStructures.h"
 
 #include "RecoTracker/MkFitCore/interface/IterationConfig.h"
-
+#include "RecoTracker/MkFitCore/src/Matriplex/Memory.h"
```
Comment (slava77) [inline RecoTracker/MkFitCore/src/HitStructures.cc:4]: we usually include the local directory private files (not from /interface) directly
```suggestion
#include "Matriplex/Memory.h"
```

**Comment by makortel** [comment]
@cmsbuild, please test for CMSSW_12_3_ASAN_X

**Comment by smuzaffar** [comment]
please test for slc7_aarch64_gcc11

lets tests the change [RecoTracker/MkFitCore/src/Matriplex/MatriplexCommon.h](https://github.com/cms-sw/cmssw/pull/37111/files#diff-00a4359bd5d8bd292e21e3540bb75c36ab445c7f01d649b4874f8221817e6f47) for non-x86_64 archs too

**Comment by slava77** [comment]
> **Failed Tests**: RelVals
> CMSSW: CMSSW_12_3_X_2022-02-28-2300/slc7_aarch64_gcc11

I suppose, as in the previous update in #37014 these workflows are failing also in the aarch IB. 
The fact that the code compiled was the main relevant part of the test.

**Comment by slava77** [comment]
> **Failed Tests**: RelVals
> **CMSSW**: CMSSW_12_3_ASAN_X_2022-02-28-1100/slc7_amd64_gcc11

it seems like only phase-2 (non-mkFit) workflows failed. Apparently the ones with mkFit completed OK.

**Comment by makortel** [comment]
@cmsbuild, please test

(regular tests too)

**Comment by clacaputo** [comment]
+reconstruction

- technical PR that addresses ASAN failure in mkFit https://github.com/cms-sw/cmssw/issues/37096
- wf with mkFit run properly in ASAN build (see https://github.com/cms-sw/cmssw/pull/37111#issuecomment-1055960850 and https://github.com/cms-sw/cmssw/pull/37111#issuecomment-1055981968)

**Comment by perrotta** [comment]
+1

**Pull request merged**